### PR TITLE
fix(modal): slot alignment

### PR DESCRIPTION
**Pull request by theJohnnyMe**
**Describe pull-request**  
Slot name convention 

**Solving issue**  
Fixes: DTS-1954

**How to test**  
1. Go to the preview link or branch out
2. Check if the names of the props are header, body, actions
3. Check if the props are working
4. Check if the header prop and/or slot are working
5. Check if "X" icons stays in place if no header text is provided

**Suggested test steps**
- [ ] Browser testing (Chrome, Safari, Firefox) 
- [ ] Keyboard operability
- [ ] Interactive elements have labels.
- [ ] Storybook controls
- [ ] Design/controls/props is aligned with other components 
- [ ] Dark/light mode and variants 
- [ ] Input fields – values should be displayed properly 
- [ ] Events

### CODE REVIEW

File: core/src/components/modal/modal.tsx [hunk 1]
```diff
@@ -139,19 +156,22 @@ export class TdsModal {
...
+            <slot name="body"></slot>
           </div>
 
-          <div class="tds-modal-actions">
-            <slot name="tds-modal-actions"></slot>
-          </div>
+          {usesActionsSlot && (
+            <div class="tds-modal-actions">
+              <slot name="actions"></slot>
+            </div>
```
Comment (mJarsater) [inline core/src/components/modal/modal.tsx:173]: Could this one be? 

```suggestion
              <slot name="actions"></slot>
```

File: core/src/components/modal/modal.tsx [hunk 2]
```diff
@@ -139,19 +156,22 @@ export class TdsModal {
           } `}
         >
           <div class="tds-modal-header">
-            <slot name="tds-modal-headline"></slot>
+            {this.header && <div class="tds-modal-header-title">{this.header}</div>}
```
Comment (mJarsater) [inline core/src/components/modal/modal.tsx:159]:
```suggestion
            {this.header && <div class="header">{this.header}</div>}
```

File: core/src/components/modal/modal.scss [hunk 3]
```diff
@@ -218,22 +218,21 @@ $modals: (
...
-.tds-modal-headline {
-  @include modal-headline;
-
-  color: var(--tds-modal-text);
+.tds-modal-header-title,
+slot[name='header']::slotted(*) {
+  @include modal-header-title;
 }
 
 .tds-modal-body {
```
Comment (mJarsater) [inline core/src/components/modal/modal.scss:238]:
```suggestion
.body {
```

File: core/src/components/modal/modal.tsx [hunk 4]
```diff
@@ -139,19 +156,22 @@ export class TdsModal {
...
+            {usesHeaderSlot && <slot name="header"></slot>}
             <button class="tds-modal-close" aria-label="close" onClick={() => this.handleClose()}>
               <tds-icon name="cross" size="20px"></tds-icon>
             </button>
           </div>
 
           <div class="tds-modal-body">
```
Comment (mJarsater) [inline core/src/components/modal/modal.tsx:166]:
```suggestion
          <div class="body">
```

Comment (mJarsater) [reply core/src/components/modal/modal.tsx:166]: Same as below. Just to not use prefixes if we don't have to.

File: core/src/components/modal/modal.tsx [hunk 5]
```diff
@@ -139,19 +156,22 @@ export class TdsModal {
           } `}
         >
           <div class="tds-modal-header">
```
Comment (mJarsater) [inline core/src/components/modal/modal.tsx:158]:
```suggestion
          <div class="header">
```

File: core/src/components/modal/modal.scss [hunk 6]
```diff
@@ -218,22 +218,21 @@ $modals: (
...
+    slot[name='actions']::slotted(*) {
       display: flex;
     }
   }
 }
 
-/* MODAL SUBELEMENTS */
+/* MODAL SUB ELEMENTS */
 
 .tds-modal-header {
```
Comment (mJarsater) [inline core/src/components/modal/modal.scss:229]:
```suggestion
.header {
```

File: core/src/components/modal/modal.tsx (hunk 1 above)
Comment (theJohnnyMe) [reply core/src/components/modal/modal.tsx:173]: I think that would work, good suggestion, might need to tweak CSS a bit, but not a big problem.

**Comment by theJohnnyMe** [comment]
- Shorter CSS classes updated
- Docs updated

**Review by mJarsater — APPROVED** [review]
🚀

**Review by alexandereneroth — APPROVED** [review]
Nice!

**Pull request merged**